packages/experimental/src/smart-session/helper/index.ts

```diff
@@ -0,0 +1,122 @@
+import { ZodError } from 'zod'
+import { ERROR_MESSAGES, SmartSessionGrantPermissionsRequestSchema } from '../schema/index.js'
+import {
+  type KeyType,
+  type SmartSessionGrantPermissionsRequest,
+  type WalletGrantPermissionsResponse
+} from '../utils/TypeUtils.js'
+
+export function validateRequest(request: SmartSessionGrantPermissionsRequest) {
+  try {
+    if (request === null || typeof request !== 'object') {
+      throw new Error(ERROR_MESSAGES.INVALID_REQUEST)
+    }
+
+    // Check for undefined chainId explicitly
+    if (!('chainId' in request) || request.chainId === undefined) {
+      throw new Error(ERROR_MESSAGES.INVALID_CHAIN_ID_TYPE)
+    }
```

Comment:
could we just use zod to do the validation?

Reply:
yes, removed.